public/app/plugins/panel/barchart/BarChartPanel.tsx

```diff
@@ -220,16 +253,38 @@ export const BarChartPanel: React.FunctionComponent<Props> = ({ data, options, w
       height={height}
     >
       {(config, alignedFrame) => {
+        if (oldConfig.current !== config) {
+          oldConfig.current = setupConfig({
+            config,
+            onUPlotClick,
+            setFocusedSeriesIdx,
+            setFocusedPointIdx,
+            setCoords,
```

Comment:
Do we really need to have these managed as three distinct options?

Can the seriesIdx, pointIdx, and corrds all be properties of the hover event?  Then it is just a single callback

Reply:
this is what i was eluding to on slack. i think internally they mapped to individual `set*()` react hooks, but calling each does a full re-render of the tooltip anyways the way it's coded now (i think), so probably this is a deoptimization since hook set calls are not coalesced until React 18.

Reply:
https://github.com/grafana/grafana/blob/935059a37648ae114d6754b59c4f5b89efee4865/public/app/plugins/panel/state-timeline/utils.ts#L114-L163

Reply:
Tried switching over to be included inside hover callback. On initial review it seems like there is a significant loss of performance due to how much the setSeries / Point index function is called inside of the tooltip interpolator, and having all the state inside the hover object causes a ton of unnecessary re-renders of the entire tooltip vs. leaving tooltip in place and just updating tooltip content